pkg/sql/table.go

```diff
@@ -776,6 +776,91 @@ func (p *planner) writeTableDesc(
 	return p.txn.Run(ctx, b)
 }
 
+// squashNewMutations rewrites the mutations list, removing
+// mutations where schema elements are added and dropped in
+// the same transaction, and marking the mutation jobs as
+// succeeded.
```

Comment:
I believe this should only care about DROP mutations in the DELETE_AND_WRITE_ONLY state that were not present in the ClusterVersion.